db/document.go

```diff
@@ -334,7 +335,7 @@ func unmarshalDocument(docid string, data []byte) (*Document, error) {
 	return doc, nil
 }
 
-func unmarshalDocumentWithXattr(docid string, data []byte, xattrData []byte, cas uint64, unmarshalLevel DocumentUnmarshalLevel) (doc *Document, err error) {
+func unmarshalDocumentWithXattr(docid string, data []byte, xattrData []byte, userXattrData []byte, cas uint64, unmarshalLevel DocumentUnmarshalLevel) (doc *Document, err error) {
```

Comment:
It's a bit counter-intuitive that unmarshalDocumentWithXattr doesn't unmarshal the user xattr if found.  If that's by design, can you remind me of the details why?  (and maybe stick a comment in the code explaining why)

Reply:
Well I currently only unmarshal at the point where we're going to need it in the sync function. I agree it seems a little counter-intuitive but this avoids any unnecessary work I think. There's no need to unmarshal until that point. In the event that an error occurs before that point I think its advantageous to not do that extra work.

Reply:
Makes sense.